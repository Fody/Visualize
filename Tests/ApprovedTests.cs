using ApprovalTests;
using ApprovalTests.Namers;
using ApprovalTests.Reporters;
using NUnit.Framework;

[UseReporter(typeof(DiffReporter))]
#if(DEBUG)
[UseApprovalSubdirectory("results-debug")]
#else
[UseApprovalSubdirectory("results-release")]
#endif
[TestFixture]
public class ApprovedTests
{
    [Test]
    public void ClassWithProperties()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPath, "AssemblyToProcess.ClassWithProperties"));
    }

    [Test]
    public void ClassWithConst()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPath, "AssemblyToProcess.ClassWithConst"));
    }

    [Test]
    public void ClassWithIndexor()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPath, "AssemblyToProcess.ClassWithIndexor"));
    }

    [Test]
    public void ClassWithDataAnnotations()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPath, "AssemblyToProcess.ClassWithDataAnnotations"));
    }

    [Test]
    public void ClassWithIEnumerable()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPath, "AssemblyToProcess.ClassWithIEnumerable"));
    }

    [Test]
    public void ClassWithICollection()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPath, "AssemblyToProcess.ClassWithICollection"));
    }

    [Test]
    public void GenericClassWithIEnumerable()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPath, "AssemblyToProcess.GenericClassWithIEnumerable`1"));
    }

    [Test]
    public void GenericClassWithICollection()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPath, "AssemblyToProcess.GenericClassWithICollection`1"));
    }

    [Test]
    public void ClassWithIEnumerableNotAttributed()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPath, "AssemblyToProcess.ClassWithIEnumerableNotAttributed"));
    }

    //[Test]
    //public void Sample()
    //{
    //    var sampleClassType = AssemblyWeaver.Assembly.GetType("AssemblyToProcess.ClassWithIEnumerable");
    //    var sample = Activator.CreateInstance(sampleClassType);
    //}
}