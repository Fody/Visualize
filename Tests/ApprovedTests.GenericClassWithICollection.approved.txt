.class public auto ansi beforefieldinit AssemblyToProcess.GenericClassWithICollection`1<T>
       extends [mscorlib]System.Object
       implements class [mscorlib]System.Collections.Generic.ICollection`1<!T>,
                  class [mscorlib]System.Collections.Generic.IEnumerable`1<!T>,
                  [mscorlib]System.Collections.IEnumerable
{
  .custom instance void [mscorlib]System.Diagnostics.DebuggerTypeProxyAttribute::.ctor(class [mscorlib]System.Type) = ( 01 00 52 41 73 73 65 6D 62 6C 79 54 6F 50 72 6F   // ..RAssemblyToPro
                                                                                                                        63 65 73 73 2E 47 65 6E 65 72 69 63 43 6C 61 73   // cess.GenericClas
                                                                                                                        73 57 69 74 68 49 43 6F 6C 6C 65 63 74 69 6F 6E   // sWithICollection
                                                                                                                        60 31 2B 3C 47 65 6E 65 72 69 63 43 6C 61 73 73   // `1+<GenericClass
                                                                                                                        57 69 74 68 49 43 6F 6C 6C 65 63 74 69 6F 6E 3E   // WithICollection>
                                                                                                                        50 72 6F 78 79 00 00 )                            // Proxy..
  .custom instance void [mscorlib]System.Diagnostics.DebuggerDisplayAttribute::.ctor(string) = ( 01 00 2B 43 6F 75 6E 74 20 3D 20 7B 43 6F 75 6E   // ..+Count = {Coun
                                                                                                 74 7D 20 7C 20 49 73 52 65 61 64 4F 6E 6C 79 20   // t} | IsReadOnly 
                                                                                                 3D 20 7B 49 73 52 65 61 64 4F 6E 6C 79 7D 00 00 ) // = {IsReadOnly}..
  .class auto ansi sealed nested private beforefieldinit '<GetEnumerator>d__0'<T>
         extends [mscorlib]System.Object
         implements class [mscorlib]System.Collections.Generic.IEnumerator`1<!T>,
                    [mscorlib]System.Collections.IEnumerator,
                    [mscorlib]System.IDisposable
  {
    .custom instance void [mscorlib]System.Runtime.CompilerServices.CompilerGeneratedAttribute::.ctor() = ( 01 00 00 00 ) 
    .field private !T '<>2__current'
    .field private int32 '<>1__state'
    .field public class AssemblyToProcess.GenericClassWithICollection`1<!T> '<>4__this'
    .method private hidebysig newslot virtual final 
            instance bool  MoveNext() cil managed
    {
      .override [mscorlib]System.Collections.IEnumerator::MoveNext
      // Code size       33 (0x21)
      .maxstack  2
      .locals init ([0] bool CS$1$0000,
               [1] int32 CS$4$0001)
      .language '{3F5162F8-07C6-11D3-9053-00C04FA302A1}', '{994B45C4-E6E9-11D2-903F-00C04FA302A1}', '{5A869D0B-6611-11D3-BD2A-0000F80849BD}'
      .line 16707566,16707566 : 0,0 'AssemblyToProcess\\GenericClassWithICollection.cs'
      IL_0000:  ldarg.0
      IL_0001:  ldfld      int32 class AssemblyToProcess.GenericClassWithICollection`1/'<GetEnumerator>d__0'<!T>::'<>1__state'
      IL_0006:  stloc.1
      IL_0007:  ldloc.1
      IL_0008:  ldc.i4.0
      IL_0009:  beq.s      IL_000d
      IL_000b:  br.s       IL_000f
      IL_000d:  br.s       IL_0011
      IL_000f:  br.s       IL_001b
      IL_0011:  ldarg.0
      IL_0012:  ldc.i4.m1
      IL_0013:  stfld      int32 class AssemblyToProcess.GenericClassWithICollection`1/'<GetEnumerator>d__0'<!T>::'<>1__state'
      .line 12,12 : 9,10 ''
      IL_0018:  nop
      .line 13,13 : 13,25 ''
      IL_0019:  br.s       IL_001b
      .line 16707566,16707566 : 0,0 ''
      IL_001b:  ldc.i4.0
      IL_001c:  stloc.0
      IL_001d:  br.s       IL_001f
      IL_001f:  ldloc.0
      IL_0020:  ret
    } // end of method '<GetEnumerator>d__0'::MoveNext
    .method private hidebysig newslot specialname virtual final 
            instance !T  'System.Collections.Generic.IEnumerator<T>.get_Current'() cil managed
    {
      .custom instance void [mscorlib]System.Diagnostics.DebuggerHiddenAttribute::.ctor() = ( 01 00 00 00 ) 
      .override  method instance !0 class [mscorlib]System.Collections.Generic.IEnumerator`1<!T>::get_Current()
      // Code size       11 (0xb)
      .maxstack  1
      .locals init (!T V_0)
      IL_0000:  ldarg.0
      IL_0001:  ldfld      !0 class AssemblyToProcess.GenericClassWithICollection`1/'<GetEnumerator>d__0'<!T>::'<>2__current'
      IL_0006:  stloc.0
      IL_0007:  br.s       IL_0009
      IL_0009:  ldloc.0
      IL_000a:  ret
    } // end of method '<GetEnumerator>d__0'::'System.Collections.Generic.IEnumerator<T>.get_Current'
    .method private hidebysig newslot virtual final 
            instance void  System.Collections.IEnumerator.Reset() cil managed
    {
      .custom instance void [mscorlib]System.Diagnostics.DebuggerHiddenAttribute::.ctor() = ( 01 00 00 00 ) 
      .override [mscorlib]System.Collections.IEnumerator::Reset
      // Code size       6 (0x6)
      .maxstack  8
      IL_0000:  newobj     instance void [mscorlib]System.NotSupportedException::.ctor()
      IL_0005:  throw
    } // end of method '<GetEnumerator>d__0'::System.Collections.IEnumerator.Reset
    .method private hidebysig newslot virtual final 
            instance void  System.IDisposable.Dispose() cil managed
    {
      .override [mscorlib]System.IDisposable::Dispose
      // Code size       2 (0x2)
      .maxstack  8
      IL_0000:  nop
      .line 16707566,16707566 : 0,0 ''
      IL_0001:  ret
    } // end of method '<GetEnumerator>d__0'::System.IDisposable.Dispose
    .method private hidebysig newslot specialname virtual final 
            instance object  System.Collections.IEnumerator.get_Current() cil managed
    {
      .custom instance void [mscorlib]System.Diagnostics.DebuggerHiddenAttribute::.ctor() = ( 01 00 00 00 ) 
      .override [mscorlib]System.Collections.IEnumerator::get_Current
      // Code size       16 (0x10)
      .maxstack  1
      .locals init (object V_0)
      IL_0000:  ldarg.0
      IL_0001:  ldfld      !0 class AssemblyToProcess.GenericClassWithICollection`1/'<GetEnumerator>d__0'<!T>::'<>2__current'
      IL_0006:  box        !T
      IL_000b:  stloc.0
      IL_000c:  br.s       IL_000e
      IL_000e:  ldloc.0
      IL_000f:  ret
    } // end of method '<GetEnumerator>d__0'::System.Collections.IEnumerator.get_Current
    .method public hidebysig specialname rtspecialname 
            instance void  .ctor(int32 '<>1__state') cil managed
    {
      .custom instance void [mscorlib]System.Diagnostics.DebuggerHiddenAttribute::.ctor() = ( 01 00 00 00 ) 
      // Code size       14 (0xe)
      .maxstack  8
      IL_0000:  ldarg.0
      IL_0001:  call       instance void [mscorlib]System.Object::.ctor()
      IL_0006:  ldarg.0
      IL_0007:  ldarg.1
      IL_0008:  stfld      int32 class AssemblyToProcess.GenericClassWithICollection`1/'<GetEnumerator>d__0'<!T>::'<>1__state'
      IL_000d:  ret
    } // end of method '<GetEnumerator>d__0'::.ctor
    .property instance !T 'System.Collections.Generic.IEnumerator<T>.Current'()
    {
      .get instance !T AssemblyToProcess.GenericClassWithICollection`1/'<GetEnumerator>d__0'::'System.Collections.Generic.IEnumerator<T>.get_Current'()
    } // end of property '<GetEnumerator>d__0'::'System.Collections.Generic.IEnumerator<T>.Current'
    .property instance object System.Collections.IEnumerator.Current()
    {
      .get instance object AssemblyToProcess.GenericClassWithICollection`1/'<GetEnumerator>d__0'::System.Collections.IEnumerator.get_Current()
    } // end of property '<GetEnumerator>d__0'::System.Collections.IEnumerator.Current
  } // end of class '<GetEnumerator>d__0'
  .class auto ansi sealed nested private beforefieldinit '<GenericClassWithICollection>Proxy'<T>
         extends [mscorlib]System.Object
  {
    .custom instance void [mscorlib]System.Runtime.CompilerServices.CompilerGeneratedAttribute::.ctor() = ( 01 00 00 00 ) 
    .field private initonly class AssemblyToProcess.GenericClassWithICollection`1<!T> original
    .method public hidebysig specialname rtspecialname 
            instance void  .ctor(class AssemblyToProcess.GenericClassWithICollection`1<!T> original) cil managed
    {
      // Code size       14 (0xe)
      .maxstack  8
      IL_0000:  ldarg.0
      IL_0001:  call       instance void [mscorlib]System.Object::.ctor()
      IL_0006:  ldarg.0
      IL_0007:  ldarg.1
      IL_0008:  stfld      class AssemblyToProcess.GenericClassWithICollection`1<!0> class AssemblyToProcess.GenericClassWithICollection`1/'<GenericClassWithICollection>Proxy'<!T>::original
      IL_000d:  ret
    } // end of method '<GenericClassWithICollection>Proxy'::.ctor
    .method public hidebysig specialname 
            instance !T[]  get_Items() cil managed
    {
      // Code size       34 (0x22)
      .maxstack  3
      .locals init (!T[] V_0)
      IL_0000:  ldarg.0
      IL_0001:  ldfld      class AssemblyToProcess.GenericClassWithICollection`1<!0> class AssemblyToProcess.GenericClassWithICollection`1/'<GenericClassWithICollection>Proxy'<!T>::original
      IL_0006:  callvirt   instance int32 class AssemblyToProcess.GenericClassWithICollection`1<!T>::get_Count()
      IL_000b:  newarr     !T
      IL_0010:  stloc.0
      IL_0011:  ldarg.0
      IL_0012:  ldfld      class AssemblyToProcess.GenericClassWithICollection`1<!0> class AssemblyToProcess.GenericClassWithICollection`1/'<GenericClassWithICollection>Proxy'<!T>::original
      IL_0017:  ldloc.0
      IL_0018:  ldloc.0
      IL_0019:  ldlen
      IL_001a:  conv.i4
      IL_001b:  callvirt   instance void class AssemblyToProcess.GenericClassWithICollection`1<!T>::CopyTo(!0[],
                                                                                                           int32)
      IL_0020:  ldloc.0
      IL_0021:  ret
    } // end of method '<GenericClassWithICollection>Proxy'::get_Items
    .property instance !T[] Items()
    {
      .custom instance void [mscorlib]System.Diagnostics.DebuggerBrowsableAttribute::.ctor(valuetype [mscorlib]System.Diagnostics.DebuggerBrowsableState) = ( 01 00 03 00 00 00 00 00 ) 
      .get instance !T[] AssemblyToProcess.GenericClassWithICollection`1/'<GenericClassWithICollection>Proxy'::get_Items()
    } // end of property '<GenericClassWithICollection>Proxy'::Items
  } // end of class '<GenericClassWithICollection>Proxy'
  .method public hidebysig newslot virtual final 
          instance class [mscorlib]System.Collections.Generic.IEnumerator`1<!T> 
          GetEnumerator() cil managed
  {
    // Code size       20 (0x14)
    .maxstack  2
    .locals init (class AssemblyToProcess.GenericClassWithICollection`1/'<GetEnumerator>d__0'<!T> V_0,
             class [mscorlib]System.Collections.Generic.IEnumerator`1<!T> V_1)
    IL_0000:  ldc.i4.0
    IL_0001:  newobj     instance void class AssemblyToProcess.GenericClassWithICollection`1/'<GetEnumerator>d__0'<!T>::.ctor(int32)
    IL_0006:  stloc.0
    IL_0007:  ldloc.0
    IL_0008:  ldarg.0
    IL_0009:  stfld      class AssemblyToProcess.GenericClassWithICollection`1<!0> class AssemblyToProcess.GenericClassWithICollection`1/'<GetEnumerator>d__0'<!T>::'<>4__this'
    IL_000e:  ldloc.0
    IL_000f:  stloc.1
    IL_0010:  br.s       IL_0012
    IL_0012:  ldloc.1
    IL_0013:  ret
  } // end of method GenericClassWithICollection`1::GetEnumerator
  .method private hidebysig newslot virtual final 
          instance class [mscorlib]System.Collections.IEnumerator 
          System.Collections.IEnumerable.GetEnumerator() cil managed
  {
    .override [mscorlib]System.Collections.IEnumerable::GetEnumerator
    // Code size       12 (0xc)
    .maxstack  1
    .locals init ([0] class [mscorlib]System.Collections.IEnumerator CS$1$0000)
    .line 17,17 : 9,10 ''
    IL_0000:  nop
    .line 18,18 : 13,36 ''
    IL_0001:  ldarg.0
    IL_0002:  call       instance class [mscorlib]System.Collections.Generic.IEnumerator`1<!0> class AssemblyToProcess.GenericClassWithICollection`1<!T>::GetEnumerator()
    IL_0007:  stloc.0
    IL_0008:  br.s       IL_000a
    .line 19,19 : 9,10 ''
    IL_000a:  ldloc.0
    IL_000b:  ret
  } // end of method GenericClassWithICollection`1::System.Collections.IEnumerable.GetEnumerator
  .method public hidebysig newslot virtual final 
          instance void  Add(!T item) cil managed
  {
    // Code size       7 (0x7)
    .maxstack  8
    .line 22,22 : 9,10 ''
    IL_0000:  nop
    .line 23,23 : 13,49 ''
    IL_0001:  newobj     instance void [mscorlib]System.NotImplementedException::.ctor()
    IL_0006:  throw
  } // end of method GenericClassWithICollection`1::Add
  .method public hidebysig newslot virtual final 
          instance void  Clear() cil managed
  {
    // Code size       7 (0x7)
    .maxstack  8
    .line 27,27 : 9,10 ''
    IL_0000:  nop
    .line 28,28 : 13,49 ''
    IL_0001:  newobj     instance void [mscorlib]System.NotImplementedException::.ctor()
    IL_0006:  throw
  } // end of method GenericClassWithICollection`1::Clear
  .method public hidebysig newslot virtual final 
          instance bool  Contains(!T item) cil managed
  {
    // Code size       7 (0x7)
    .maxstack  8
    .line 32,32 : 9,10 ''
    IL_0000:  nop
    .line 33,33 : 13,49 ''
    IL_0001:  newobj     instance void [mscorlib]System.NotImplementedException::.ctor()
    IL_0006:  throw
  } // end of method GenericClassWithICollection`1::Contains
  .method public hidebysig newslot virtual final 
          instance void  CopyTo(!T[] 'array',
                                int32 arrayIndex) cil managed
  {
    // Code size       7 (0x7)
    .maxstack  8
    .line 37,37 : 9,10 ''
    IL_0000:  nop
    .line 38,38 : 13,49 ''
    IL_0001:  newobj     instance void [mscorlib]System.NotImplementedException::.ctor()
    IL_0006:  throw
  } // end of method GenericClassWithICollection`1::CopyTo
  .method public hidebysig newslot specialname virtual final 
          instance int32  get_Count() cil managed
  {
    // Code size       8 (0x8)
    .maxstack  1
    .locals init ([0] int32 CS$1$0000)
    .line 43,43 : 17,18 ''
    IL_0000:  nop
    .line 43,43 : 19,29 ''
    IL_0001:  ldc.i4.s   10
    IL_0003:  stloc.0
    IL_0004:  br.s       IL_0006
    .line 43,43 : 30,31 ''
    IL_0006:  ldloc.0
    IL_0007:  ret
  } // end of method GenericClassWithICollection`1::get_Count
  .method public hidebysig newslot specialname virtual final 
          instance bool  get_IsReadOnly() cil managed
  {
    // Code size       7 (0x7)
    .maxstack  1
    .locals init ([0] bool CS$1$0000)
    .line 48,48 : 17,18 ''
    IL_0000:  nop
    .line 48,48 : 19,31 ''
    IL_0001:  ldc.i4.1
    IL_0002:  stloc.0
    IL_0003:  br.s       IL_0005
    .line 48,48 : 32,33 ''
    IL_0005:  ldloc.0
    IL_0006:  ret
  } // end of method GenericClassWithICollection`1::get_IsReadOnly
  .method public hidebysig newslot virtual final 
          instance bool  Remove(!T item) cil managed
  {
    // Code size       7 (0x7)
    .maxstack  8
    .line 52,52 : 9,10 ''
    IL_0000:  nop
    .line 53,53 : 13,49 ''
    IL_0001:  newobj     instance void [mscorlib]System.NotImplementedException::.ctor()
    IL_0006:  throw
  } // end of method GenericClassWithICollection`1::Remove
  .method public hidebysig specialname rtspecialname 
          instance void  .ctor() cil managed
  {
    // Code size       7 (0x7)
    .maxstack  8
    IL_0000:  ldarg.0
    IL_0001:  call       instance void [mscorlib]System.Object::.ctor()
    IL_0006:  ret
  } // end of method GenericClassWithICollection`1::.ctor
  .property instance int32 Count()
  {
    .get instance int32 AssemblyToProcess.GenericClassWithICollection`1::get_Count()
  } // end of property GenericClassWithICollection`1::Count
  .property instance bool IsReadOnly()
  {
    .get instance bool AssemblyToProcess.GenericClassWithICollection`1::get_IsReadOnly()
  } // end of property GenericClassWithICollection`1::IsReadOnly
} // end of class AssemblyToProcess.GenericClassWithICollection`1